### ISSUE TITLE
`ff init -b fabric dev1` creates a non-functional installation

### ISSUE THREAD

**Issue by paul-snively** [hyperledger/firefly#537]
When I say:

`ff init -b fabric dev1`

and then:

`ff start dev1`

I get:

```
reading stack config... done
this will take a few seconds longer since this is the first time you're running this stack...
⣯ registering identities... Post "http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true": read tcp 127.0.0.1:43086->127.0⣾ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣻ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⡿ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣯ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣾ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣻ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⡿ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣯ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣾ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣻ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⡿ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣯ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣾ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣻ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⡿ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣯ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣾ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣻ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⡿ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣯ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣾ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣻ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⡿ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣯ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣾ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣻ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⡿ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣯ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⣷ registering org and node for member 0... http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
⡿ Error: an error occurred - rolling back changes... Error: http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
 - all changes rolled back
Usage:
  ff start <stack_name> [flags]

Flags:
  -h, --help          help for start
  -b, --no-rollback   Do not automatically rollback changes if first time setup fails

Global Flags:
      --ansi string   control when to print ANSI control characters ("never"|"always"|"auto") (default "auto")
  -v, --verbose       verbose log output

Error: http://127.0.0.1:5000/api/v1/network/organizations/self?confirm=true returned 500: {"error":"FF10286: Failed to decode certificate: illegal base64 data at input byte 0"}
 - all changes rolled back
```

**Comment by nguyer** [comment]
Hi @paul-snively thanks for reporting this. I think this was actually due to a change in fabconnect which was fixed in this PR earlier today: https://github.com/hyperledger/firefly-fabconnect/pull/80

**Comment by nguyer** [comment]
As of this morning, I'm unable to reproduce this issue on my local machine, using the latest CLI, and either the latest FireFly release, or "head". @paul-snively Are you still having this issue? If not I'll go ahead and close this out.

**Comment by paul-snively** [comment]
I'm honestly not sure how to check. I just ensured that I have the CLI version 0.0.44, and I still get the error. 0.0.44 was released six days ago, so I'm not sure how it could have fixed anything unless it's pulling the `latest` fabconnect, maybe?

Anyway, please advise, and thanks for your prompt attention.

**Comment by nguyer** [comment]
Okay I just did a `docker system prune -a` just to make sure I didn't have an image cached or something. After that I ran:
```
ff init -b fabric dev1 2
ff start dev1
```

This brings up a functional fabric stack for me. However... in investigating this I did discover a bug in the fabric blockchain provider code in the FireFly CLI. It appears to be always pulling `latest` which explains why the bug seems fixed for me, and may explain why you and I have been getting inconsistent results - because `latest` points to different images at different times.

It _should_ be pinned to a specific release SHA, but it looks like we missed the fabconnect image when we write the docker-compose file. I'll take a look at getting a fix for this in.

In the meantime, doing an `ff remove dev` and trying again should fix your original problem :)
